test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -229,13 +251,53 @@ TEST_F(QuicPlatformTest, QuicMockLog) {
 }
 
 TEST_F(QuicPlatformTest, QuicServerStats) {
-  // Just make sure they compile.
-  QUIC_SERVER_HISTOGRAM_ENUM("my.enum.histogram", TestEnum::ONE, TestEnum::COUNT, "doc");
-  QUIC_SERVER_HISTOGRAM_BOOL("my.bool.histogram", false, "doc");
-  QUIC_SERVER_HISTOGRAM_TIMES("my.timing.histogram", QuicTime::Delta::FromSeconds(5),
-                              QuicTime::Delta::FromSeconds(1), QuicTime::Delta::FromSecond(3600),
-                              100, "doc");
-  QUIC_SERVER_HISTOGRAM_COUNTS("my.count.histogram", 123, 0, 1000, 100, "doc");
+  Envoy::Stats::Histogram& h1 = store_->histogramFromString(
+      "quiche.test_latency", Envoy::Stats::Histogram::Unit::Milliseconds);
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h1), 100));
+  QUIC_SERVER_HISTOGRAM_TIMES(test_latency, 100, 0, 500, 10, "bbb");
+
+  Envoy::Stats::Histogram& h2 = store_->histogramFromString(
+      "quiche.test_trial.success", Envoy::Stats::Histogram::Unit::Unspecified);
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h2), 0));
+  QUIC_SERVER_HISTOGRAM_BOOL(test_trial.success, false, "just for test");
```

Comment:
With out the stats struct it would look like:
```
    QuicStatsImpl& quiche_stats = getQuicheStats();
     Envoy::Stats::SymbolTable& symbol_table = quiche_stats.symbolTable();
    // Assuming we don't support : nvoy::Stats::StatNameDynamicStorage storage1("test_trial.success", symbol_table);
     Envoy::Stats::StatNameDynamicStorage storage1("test_trial", symbol_table);
     Envoy::Stats::StatNameDynamicStorage storage2("success", symbol_table);
     Envoy::Stats::SymbolTable::StoragePtr joined = symbol_table.join({
        quiche_stats.quicheStatName(), storage1.statName(), storage2.statsName});
    auto result = quiche_stats.scope().histogramFromStatName(StatName(joined.get());
    result.recordValue(false);
```

Reply:
right, and once https://github.com/envoyproxy/envoy/pull/10715 is merged this would be:
```
    QuicStatsImpl& quiche_stats = getQuicheStats();
    auto hist = Envoy::Stats::Utility::histogramFromElements(quiche_stats.scope(), {
      quiche_stats.quicheStatName(), "test_trial.success"});
    hist.recordValue(false);
```
Though it could be simplified further; if you are always prefixing with "quiche" then when constructing QuicheStatsImpl, you would just make a nested scope in your ctor:
```
   : scope_(scope.createScope("quiche")) { ... }
```
and use that. Then it simplifies to one line if you really want.
```
Envoy::Stats::Utility::histogramFromElements(getQuicheStats().scope(), {"test_trial.success"}).recordValue(false);
```
though we really want to do factor out any common components from your dot-separated names and save them on construction in the QuicheStatsImpl.

And I guess if I bit the bullet and made histogramFromElements a Scope method then it would be a little cleaner still:
```
getQuicheStats().scope().histogramFromElements({"test_trial.success"}).recordValue(false);
```

Reply:
What's the difference between  ThreadLocalStore::histogramFromString(name) and ThreadLocalStore::histogramFromElements(name) then? Or I'm wondering how to choose between these two interface when we instantiate or search for histograms?